### DISCUSSION TITLE
Heroku and Redis

### DISCUSSION THREAD

**Discussion by mrswats** [python-telegram-bot/python-telegram-bot#2914] [Q&A]
I made a bot that calls a public API and the bot will display the data gathered by this API. However, since it is sunset and sunraise data, I do not need to call the API every time so I figured I would use redis to store the data and this way I won't have to call the API every time.

However, everything seems to work great but when I try to setup the webhook in heroku I get the following error:
```
telegram.ext.updater -- [ERROR] 2022-02-23 07:02:20,144: Error while bootstrap set webhook: Bad webhook: failed to resolve host: no address associated with hostname
telegram.ext.updater -- [ERROR] 2022-02-23 07:02:20,145: Failed bootstrap phase after 0 retries (Bad webhook: failed to resolve host: no address associated with hostname)
telegram.ext.updater -- [ERROR] 2022-02-23 07:02:20,145: unhandled exception in Bot:5143305038:updater
```

I have set up the bot for heroku as shown in the wiki like:

```
updater = Updater(TELEGRAM_API_TOKEN)
updater.start_webhook(
    listen="0.0.0.0",
    port=PORT,
    url_path=TELEGRAM_API_TOKEN,
    webhook_url=f"{WEBHOOK_URL}/{TELEGRAM_API_TOKEN}",
)
updater.idle()
```
where `PORT` is the port that heroku gives me and `WEBHOOK_URL` is the heroku app URL provided when creating the heroku app. I have deployed other bots on heroku using this same method without any problems, but this one doesn't want to work. I am guessing that the redis instance has __something__ to do with all that, but I do not know what is the correct setup for this.

Thank you in advance.

**Comment by Bibo-Joshi** [comment]
The error messages don't look like it has anything to do with redis. You can try to remove all redis-related logic from your code to double check.
